src/webview/openshift-terminal/openShiftTerminal.ts

```diff
@@ -0,0 +1,588 @@
+    private _onExitListener: () => void;
+    private _onTextListener: (text: string) => void;
+
+    private _closeOnExit: boolean;
+
+    private _uuid: string;
+    private _headlessTerm: Terminal;
+    private _termSerializer: SerializeAddon;
+    private _disposables: { dispose(): void }[];
+    private _ptyExited = false;
+
+    private _terminalRendering = false;
+
+    private _buffer: Buffer;
+
+    /**
+     * Creates a new OpenShiftTerminal
+     *
+     * @param uuid a unique identifier for the terminal
+     * @param sendMessage the function used to send a message to the UI terminal
+     * @param file the path to the program to execute
+     * @param args the arguments to pass to the program
+     * @param options the options for spawning the pty (name, current working directory, environment)
+     * @param closeOnExit true if the terminal should close when the program exits, or false if it should stay open until the user presses an additional key
+     * @param callbacks functions to execute in response to events in the terminal:
+     * - onSpawn(): called when the pty is created
+     * - onExit(): called when the pty exits (whether normally or though SIGABRT)
+     * - onText(): called when text is printed to the terminal, whether by the program or by user input
+     */
+    constructor(
+        uuid: string,
+        sendMessage: (message: Message) => Promise<void>,
+        file: string,
+        args: string | string[],
+        options,
+        closeOnExit = false,
+        callbacks?: {
+            onSpawn?: () => void;
+            onExit?: () => void;
+            onText?: (text: string) => void;
+        },
+    ) {
+        this._uuid = uuid;
+
+        this._sendTerminalData = (data) => {
+            void sendMessage({ kind: 'termOutput', data: { uuid, output: data } });
+        };
+        this._sendExitMessage = () => {
+            void sendMessage({ kind: 'termExit', data: { uuid } });
+        };
+        this._onSpawnListener = callbacks?.onSpawn || (() => undefined);
+        this._onExitListener = callbacks?.onExit || (() => undefined);
+        this._onTextListener = callbacks?.onText || ((_text: string) => undefined);
+
+        this._closeOnExit = closeOnExit;
+
+        this._file = file;
+        this._args = args;
+        this._options = options;
+
+        this._disposables = [];
+        this._headlessTerm = new Terminal({ allowProposedApi: true });
+        this._termSerializer = new SerializeAddon();
+        this._headlessTerm.loadAddon(this._termSerializer);
+        this._buffer = Buffer.from('');
+
+        this._disposables.push(this._headlessTerm);
+        this._disposables.push(this._termSerializer);
+    }
+
+    startPty() {
+        if (platform() === 'win32') {
+            const escapedArgs = Array.isArray(this._args)
+                ? this._args.join(' ')
+                : this._args;
+            this._pty = ptyInstance.spawn(
+                'C:\\WINDOWS\\system32\\cmd.EXE',
```

Comment:
I'm not sure, but is it possible to use `COMSPEC` System Environment Variable here, so it  correctly finds `cmd.exe`?

> C:\Windows\System32\cmd.exe or if running a 32 bit [WOW](https://ss64.com/nt/syntax-64bit.html) - C:\Windows\SysWOW64\cmd.exe
> Comspec is used whenever the command shell spawns a new process, e.g. the [FOR](https://ss64.com/nt/for.html) command, if comspec is not defined those processes will fail to launch.

Reply:
I'll try it out...

Reply:
looks like the library I'm using contains a mechanism to resolve the location of cmd.exe, so I don't need to provide an absolute path